src/lib/components/FileUploader/FileUploader.js

```diff
@@ -4,125 +4,132 @@
 //
 // React-Invenio-Deposit is free software; you can redistribute it and/or modify it
 // under the terms of the MIT License; see LICENSE file for more details.
-import React, { Component } from 'react';
+import { useFormikContext } from 'formik';
+import React from 'react';
 import PropTypes from 'prop-types';
 import { Grid, Icon, Message } from 'semantic-ui-react';
 
 import { FileUploaderArea } from './FileUploaderArea';
 import { FileUploaderToolbar } from './FileUploaderToolbar';
 import { UploadState } from '../../state/reducers/files';
 
-export default class FileUploader extends Component {
-  render() {
-    const {
-      files,
-      record,
-      uploadFilesToDraft,
-      quota,
-      config,
-      filesEnabled,
-      isDraftRecord,
-      setFilesEnabled,
-    } = this.props;
-    let filesList = Object.values(files).map((fileState) => {
-      return {
-        name: fileState.name,
-        size: fileState.size,
-        checksum: fileState.checksum,
-        links: fileState.links,
-        upload: {
-          initial: fileState.status === UploadState.initial,
-          failed: fileState.status === UploadState.error,
-          ongoing: fileState.status === UploadState.uploading,
-          finished: fileState.status === UploadState.finished,
-          pending: fileState.status === UploadState.pending,
-          progress: fileState.progress,
-          cancel: fileState.cancel,
-        },
-      };
-    });
-    const filesSize = filesList.reduce(
-      (totalSize, file) => (totalSize += file.size),
-      0
-    );
 
-    let dropzoneParams = {
-      preventDropOnDocument: true,
-      onDropAccepted: (acceptedFiles) => {
-        const maxFileNumberReached =
-          filesList.length + acceptedFiles.length > quota.maxFiles;
-        const acceptedFilesSize = acceptedFiles.reduce(
-          (totalSize, file) => (totalSize += file.size),
-          0
-        );
-        const maxFileStorageReached =
-          filesSize + acceptedFilesSize > quota.maxStorage;
-        if (maxFileNumberReached || maxFileStorageReached) {
-          // TODO: Give some feedback to user about why the files weren't
-          // accepted for upload
-          console.log('Maximum number of files reached!');
-        } else {
-          uploadFilesToDraft(record, acceptedFiles);
-        }
+// NOTE: This component has to be a function component to allow
+//       the `useFormikContext` hook.
+export const FileUploaderComponent = ({
+  config,
+  defaultFilePreview,
+  files,
+  filesEnabled,
+  isDraftRecord,
+  quota,
+  record,
+  setFilesEnabled,
+  uploadFilesToDraft,
+  ...uiProps
+}) => {
+  // We extract the working copy of the draft stored as `values` in formik
+  const { values: formikDraft } = useFormikContext();
```

Comment:
Just an observation: We have now a mixed way of getting the values from formik i.e hook vs class approach...We should try to converge to one if possible...

Reply:
yeah... the difference I see is that in some cases we have actual fields that will need to display errors and are "real" inputs to the metadata form and in other cases we have fields that need access to the metadata form but are not inputs to the form (don't need to have other field properties). Agreed that thinking about this and establishing good practices will help.